### PR TITLE
SALTO-6312: Support Zendesk queues

### PR DESCRIPTION
**Pull request by tamtamirr**
Add support for queues

---

_Additional context for reviewer_

---
_Release Notes_: 
__Zendesk Adapter:__
* Add support for custom omnichannel routing queues

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/field_references.ts [hunk 1]
```diff
@@ -986,15 +1001,6 @@ const commonFieldNameToTypeMappingDefs: ZendeskFieldReferenceDefinition[] = [
...
 const secondIterationFieldNameToTypeMappingDefs: ZendeskFieldReferenceDefinition[] = [
-  {
-    src: {
-      field: 'value',
-      parentTypes: ['trigger__conditions__all'],
-    },
-    serializationStrategy: 'ticketFieldOption',
-    target: { typeContext: 'neighborReferenceTicketField' },
-    zendeskMissingRefStrategy: 'typeAndValue',
-  },
```
Comment (adfineberg) [inline packages/zendesk-adapter/src/filters/field_references.ts:997]: this was contained in the one below

File: packages/zendesk-adapter/src/config.ts [hunk 2]
```diff
@@ -1168,6 +1168,39 @@ export const DEFAULT_TYPES: ZendeskApiConfig['types'] = {
       },
     },
   },
+  queue: {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/config.ts:1171]: don't you need to add this to the supported types?

File: packages/zendesk-adapter/src/definitions/fetch/fetch.ts [hunk 3]
```diff
@@ -890,6 +890,29 @@ const createCustomizations = (): Record<
     },
   },
 
+  queue: {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/definitions/fetch/fetch.ts:893]: what about the pagination? did you check that?

File: packages/zendesk-adapter/src/definitions/fetch/transforms/queue_adjuster.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+import { definitions } from '@salto-io/adapter-components'
+import { values as lowerdashValues } from '@salto-io/lowerdash'
+import _ from 'lodash'
+
+// this transformer updates the group objects for queues into the correct format
+export const transform: definitions.AdjustFunctionSingle = async ({ value }) => {
+  if (!lowerdashValues.isPlainObject(value)) {
+    throw new Error('unexpected value for guide item, not transforming')
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/definitions/fetch/transforms/queue_adjuster.ts:16]:
```suggestion
    throw new Error('unexpected value for queue item, not transforming')
```

File: packages/zendesk-adapter/src/definitions/fetch/transforms/queue_adjuster.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  const primaryGroupIds = _.map(_.get(value, 'primary_groups.groups'), 'id')
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/definitions/fetch/transforms/queue_adjuster.ts:19]: what happens if `_.get(value, 'primary_groups.groups')` returns undefined? will `_.map` crush?

**Review by edenhassid — COMMENTED** [review]
LGTM! ⭐  added some small comments

File: packages/zendesk-adapter/src/definitions/fetch/transforms/queue_adjuster.ts (hunk 5 above)
Comment (adfineberg) [reply packages/zendesk-adapter/src/definitions/fetch/transforms/queue_adjuster.ts:19]: nope, works like a charm. `map` is smart like that `expect(_.map(undefined, 'id')).toEqual([])`

File: packages/zendesk-adapter/src/config.ts (hunk 2 above)
Comment (adfineberg) [reply packages/zendesk-adapter/src/config.ts:1171]: The new infra doesn't use that

**Review by edenhassid — APPROVED** [review]
nice! 💯

**Pull request merged**